integration-cli/docker_api_containers_test.go

```diff
@@ -684,33 +687,92 @@ func (s *DockerSuite) TestContainerApiCreateWithHostName(c *check.C) {
 	if err != nil && !strings.Contains(err.Error(), "200 OK: 201") {
 		c.Fatal(err)
 	}
-	type createResp struct {
-		Id string
+
+	var container types.ContainerCreateResponse
+	if err := json.Unmarshal(b, &container); err != nil {
+		c.Fatal(err)
 	}
-	var container createResp
+
+	var id = container.ID
+
+	_, bodyGet, err := sockRequest("GET", "/containers/"+id+"/json", nil)
+
+	var containerInfo types.ContainerJSON
+	if err := json.Unmarshal(bodyGet, &containerInfo); err != nil {
+		c.Fatal(err)
+	}
+	var hostNameActual = containerInfo.Config.Hostname
+	if hostNameActual != "test-host" {
+		c.Fatalf("Mismatched Hostname, Expected %v, Actual: %v ", hostName, hostNameActual)
+	}
+}
+
+func (s *DockerSuite) TestContainerApiCreateWithDomainName(c *check.C) {
+	var domainName = "test-domain"
+	config := map[string]interface{}{
+		"Image":      "busybox",
+		"Domainname": domainName,
+	}
+
+	_, b, err := sockRequest("POST", "/containers/create", config)
+	if err != nil && !strings.Contains(err.Error(), "200 OK: 201") {
+		c.Fatal(err)
+	}
+
+	var container types.ContainerCreateResponse
 	if err := json.Unmarshal(b, &container); err != nil {
 		c.Fatal(err)
 	}
 
-	var id = container.Id
+	var id = container.ID
 
 	_, bodyGet, err := sockRequest("GET", "/containers/"+id+"/json", nil)
 
-	type configLocal struct {
-		Hostname string
+	var containerInfo types.ContainerJSON
+	if err := json.Unmarshal(bodyGet, &containerInfo); err != nil {
+		c.Fatal(err)
 	}
-	type getResponse struct {
-		Id     string
-		Config configLocal
+	var domainNameActual = containerInfo.Config.Domainname
+	if domainNameActual != domainName {
+		c.Fatalf("Mismatched Hostname, Expected %v, Actual: %v ", domainName, domainNameActual)
+	}
+}
+
+func (s *DockerSuite) TestContainerApiCreateNetworkMode(c *check.C) {
+	UtilCreateNetworkMode(c, "host")
+	UtilCreateNetworkMode(c, "bridge")
+	UtilCreateNetworkMode(c, "container:web1")
+}
+
+func UtilCreateNetworkMode(c *check.C, networkMode string) {
+	config := map[string]interface{}{
+		"Image":      "busybox",
+		"HostConfig": map[string]interface{}{"NetworkMode": networkMode},
 	}
 
-	var containerInfo getResponse
+	_, b, err := sockRequest("POST", "/containers/create", config)
+
+	if err != nil && !strings.Contains(err.Error(), "200 OK: 201") {
```

Comment:
you don't need string.Contains if you check status like `c.Assert(status, check.Equal, http.StatusOK)`